src/index.ts

```diff
@@ -1,173 +1,231 @@
-  options = supplementOptions(copyOptions(options), vlSpec);
-  start(vgView, copyOptions(options));
+/**
+ * Format the value to be shown in the toolip.
+ *
+ * @param value The value to show in the tooltip.
+ */
+function formatValue(value: any): string {
+  if (isArray(value)) {
+    return `[${value.map(v => (isString(v) ? v : stringify(v))).join(', ')}]`;
+  }
 
-  return {
-    destroy: function () {
-      // remove event listeners
-      vgView.removeEventListener('mouseover.tooltipInit');
-      vgView.removeEventListener('mousemove.tooltipUpdate');
-      vgView.removeEventListener('mouseout.tooltipRemove');
+  if (isString(value)) {
+    return value;
+  }
 
-      cancelPromise(); // clear tooltip promise
-    }
-  };
-}
+  if (isObject(value)) {
+    let content = '';
 
-function start(vgView: VgView, options: Option) {
-  // initialize tooltip with item data and options on mouse over
-  vgView.addEventListener('mouseover.tooltipInit', function (event: MouseEvent, item: Scenegraph) {
-    if (shouldShowTooltip(item)) {
-      // clear existing promise because mouse can only point at one thing at a time
-      cancelPromise();
-
-      // make a new promise with time delay for tooltip
-      tooltipPromise = window.setTimeout(function () {
-        init(event, item, options);
-      }, options.delay || DELAY);
-    }
-  });
+    const { title, ...rest } = value as any;
 
-  // update tooltip position on mouse move
-  // (important for large marks e.g. bars)
-  vgView.addEventListener('mousemove.tooltipUpdate', function (event: MouseEvent, item: Scenegraph) {
-    if (shouldShowTooltip(item) && tooltipActive) {
-      update(event, item, options);
+    if (title) {
+      content += `<h2>${title}</h2>`;
     }
-  });
-
-  // clear tooltip on mouse out
-  vgView.addEventListener('mouseout.tooltipRemove', function (event: MouseEvent, item: Scenegraph) {
-    if (shouldShowTooltip(item)) {
-      cancelPromise();
 
-      if (tooltipActive) {
-        clear(event, item, options);
+    content += '<table>';
+    for (const key of Object.keys(rest)) {
+      let val = (rest as any)[key];
+      if (isObject(val)) {
+        val = stringify(val);
       }
+
+      content += `<tr><td class="key">${key}:</td><td class="value">${val}</td></tr>`;
     }
-  });
-}
+    content += `</table>`;
+
+    return content;
+  }
 
-/* Cancel tooltip promise */
-function cancelPromise() {
-  /* We don't check if tooltipPromise is valid because passing
-   an invalid ID to clearTimeout does not have any effect
-   (and doesn't throw an exception). */
-  window.clearTimeout(tooltipPromise);
-  tooltipPromise = undefined;
+  return `${value}`;
 }
 
-/* Initialize tooltip with data */
-function init(event: MouseEvent, item: Scenegraph, options: Option): void {
-  // get tooltip HTML placeholder
-  const tooltipPlaceholder = getTooltipPlaceholder();
+/**
+ * Attach a tooltip handler to the view.
+ */
+export default function(view: View, opt?: Options) {
```

Comment:
Right now the design combines instantiation of the tooltip handler and registration with the View object within a single method. I would recommend against this. A cleaner / more modular / more reusable approach might be to separate these:

1. Have a way of instantiating the tooltip handler and getting back a reference that one could then "manually" register with a Vega View -- ideally including as input to the View constructor.
2. Provide a convenience "wrapper" method for creating the handler and registering it with the view.

This should be all the easier once you remove the mousemove listener and destroy method!